### ISSUE TITLE
Way to stop b-table from re-sorting immediately on data update?

### ISSUE THREAD

**Issue by zcoop98** [bootstrap-vue/bootstrap-vue#5812]
### Describe the ~~bug~~ question

I'm using the `b-table` component and its built-in sorting functionality. 

In each row of my data, I'm using slots to insert a checkbox, which the user can click to select or deselect the row and associated data (I chose this over the native "select row" functionality for reasons). I've added the ability to sort by this column using a custom compare function bound to the `:sort-compare` prop, allowing the table to bring all the checked rows to the top (or bottom). The problem is that if the user checks/ unchecks a row while sorted this way, the row instantly gets re-sorted, and jumps elsewhere in the table, which is very undesirable.

My question is this: Is there a way to prevent instant re-sorting, and only re-sort when the user changes the sort (when `sort-changed` fires)? Or do I need to implement manual sorting to make this happen?

### Versions

**Libraries:**

- BootstrapVue: 2.11.0
- Bootstrap: 4.4.1
- Vue: 2.6.11

**Environment:**

- Device: PC
- OS: Windows 10
- Browser: Chrome
- Version: 85.0.4183.102

**Comment by zcoop98** [comment]
In fact... looking at it now, there's some really funky behavior any time you edit data in the column that the table is currently being sorted by.

Fiddle:
https://jsfiddle.net/a2hn0c63/1/

Try sorting by `isActive` and playing with the checkboxes, specifically in the first and last rows.

Then try sorting by `age` and stepping a field up or down until it matches another input. It appears to take the other data hostage and update it too... What's going on here?

**Comment by zcoop98** [comment]
Examples in gif form:

<details>
<summary>Editing input in sorted column</summary>

![b-table-sorting-edit-crazy](https://user-images.githubusercontent.com/8434006/93644388-7da51380-f9c7-11ea-9139-b2bc2359ebb7.gif)

</details>

<details>
<summary>Editing checkboxes in sorted column</summary>

![b-table-sorting-check-edit](https://user-images.githubusercontent.com/8434006/93644574-e5f3f500-f9c7-11ea-8e62-7c589abb12fe.gif)

</details>

**Comment by zcoop98** [comment]
<details>
<summary>Here's what I mean, in gif form (click to reveal)</summary>

![b-table-instant-sort](https://user-images.githubusercontent.com/8434006/93645441-a0d0c280-f9c9-11ea-8def-9214c8371840.gif)
</details>

**Comment by zcoop98** [comment]
Feel like it's worth mentioning that by now I just went ahead and disabled sorting with the `no-local-sorting` prop, and pulled the custom sort routine that I already wrote for `sort-compare` into my own sort function. I tied this to the `@sort-changed` event, and got the desired effect: sorting only happens when you click the header, and doesn't immediately change with edited data.

It would still be nice to be able to set a boolean prop for this behavior however, as I personally didn't at all expect the immediate-re-sort behavior that's default. In my experience, most tables I've interacted with don't seem to do that; immediate re-sorting here felt jarring and unexpected.
